From 42c0071272a9b9b2f936aa222b5c2386fc4de5b7 Mon Sep 17 00:00:00 2001
From: Robby Cai <R63905@freescale.com>
Date: Wed, 22 Aug 2012 21:47:59 +0800
Subject: [PATCH 1953/2463] ENGR00221131: imx6sl arm2/evk: add mma8450q
 accelerometer support

mma8450q on E-INK DC3 boards, with i2c address 0x1c on I2C1.

Signed-off-by: Robby Cai <R63905@freescale.com>
---
 arch/arm/configs/imx6s_defconfig     |    4 ++--
 arch/arm/mach-mx6/board-mx6sl_arm2.c |    2 ++
 arch/arm/mach-mx6/board-mx6sl_evk.c  |    2 ++
 3 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm/configs/imx6s_defconfig b/arch/arm/configs/imx6s_defconfig
index 390770f..f03e456 100644
--- a/arch/arm/configs/imx6s_defconfig
+++ b/arch/arm/configs/imx6s_defconfig
@@ -1382,8 +1382,8 @@ CONFIG_SENSORS_MAX17135=y
 # CONFIG_SENSORS_W83627HF is not set
 # CONFIG_SENSORS_W83627EHF is not set
 CONFIG_SENSORS_MAG3110=y
-# CONFIG_MXC_MMA8450 is not set
-CONFIG_MXC_MMA8451=y
+CONFIG_MXC_MMA8450=y
+# CONFIG_MXC_MMA8451 is not set
 CONFIG_THERMAL=y
 # CONFIG_THERMAL_HWMON is not set
 CONFIG_WATCHDOG=y
diff --git a/arch/arm/mach-mx6/board-mx6sl_arm2.c b/arch/arm/mach-mx6/board-mx6sl_arm2.c
index 5e3d915..d1cf110 100755
--- a/arch/arm/mach-mx6/board-mx6sl_arm2.c
+++ b/arch/arm/mach-mx6/board-mx6sl_arm2.c
@@ -562,6 +562,8 @@ static struct i2c_board_info mxc_i2c0_board_info[] __initdata = {
 	}, {
 		I2C_BOARD_INFO("elan-touch", 0x10),
 		.irq = gpio_to_irq(MX6SL_BRD_ELAN_INT),
+	}, {
+		I2C_BOARD_INFO("mma8450", 0x1c),
 	},
 };
 
diff --git a/arch/arm/mach-mx6/board-mx6sl_evk.c b/arch/arm/mach-mx6/board-mx6sl_evk.c
index 7ef077f..11623cc 100644
--- a/arch/arm/mach-mx6/board-mx6sl_evk.c
+++ b/arch/arm/mach-mx6/board-mx6sl_evk.c
@@ -575,6 +575,8 @@ static struct i2c_board_info mxc_i2c0_board_info[] __initdata = {
 	}, {
 		I2C_BOARD_INFO("elan-touch", 0x10),
 		.irq = gpio_to_irq(MX6SL_BRD_ELAN_INT),
+	}, {
+		I2C_BOARD_INFO("mma8450", 0x1c),
 	},
 };
 
-- 
1.7.7.4

